File: generator/helperGetduration.m
function duration = helperGetduration(builtMotion, Ts)
    % HELPERGETDURATION  Calculate the duration of a built motion.
    %   duration = helperGetduration(builtMotion, Ts) takes a built motion struct
    %   and the sampling period Ts, and returns the total duration in seconds.
    %
    %   Inputs:
    %       builtMotion - Struct containing motion sequences (e.g., flShape).
    %       Ts - Sampling period in seconds.
    %   Output:
    %       duration - Total duration of the motion in seconds.

    % Get the number of samples from the flShape array
    numSamples = length(builtMotion.flShape);
    
    % Calculate duration as number of samples multiplied by sampling period
    duration = numSamples * Ts;
end

File: generator/helper_get_duration.py
def helper_get_duration(built_motion, Ts):
    """
    Calculate the duration of a built motion.

    Parameters:
        built_motion (dict): Dictionary containing motion sequences (e.g., 'flShape' as a list or array).
        Ts (float): Sampling period in seconds.

    Returns:
        float: Total duration of the motion in seconds.
    """
    num_samples = len(built_motion['flShape'])
    duration = num_samples * Ts
    return duration
